### PR TITLE
Refactor json serializer and prep for 2018-08-09

### PR DESCRIPTION
**Pull request by bhelx**
This has a number of changes:

1. Refactors json code to work around parsing bugs and make the implementation cleaner.
2. Upgrades Restsharp.
3. Contains latest changes for 2018-08-09.
4. Removes `mono` and nuget from the `install-deps` and `check-deps` scripts as it's no longer needed for dotnet core.
5. Fix the `clean` script and clean up files it has missed (e.g.  Recurly/Resources/ShippingService.cs)
6. Mark as 3.0.0-beta.1
7. Target netcoreapp2.2 and netstandard2.0

Will need to update dotnet for development:
```
sudo rm -rf /usr/local/share/dotnet/
./scripts/install-deps
```

### CODE REVIEW

File: Recurly/BaseClient.cs [hunk 1]
```diff
@@ -67,10 +68,8 @@ public class BaseClient {
 
             // If we have a body, serialize it and add it to the request
             if (body != null) {
-                string json = Json.Serialize(body); 
-                Debug.WriteLine("body: ");
-                Debug.WriteLine(json);
-                request.AddParameter("application/json", json , ParameterType.RequestBody);
+                //request.AddParameter("application/json", json , ParameterType.RequestBody);
```
Comment (aaron-suarez) [inline Recurly/BaseClient.cs]: Do we want to leave this commented out in the source code or just go ahead and remove it?

File: Recurly/JsonSerializer.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                NullValueHandling = NullValueHandling.Ignore,
+                MissingMemberHandling = MissingMemberHandling.Ignore,
+                DateFormatHandling = DateFormatHandling.IsoDateFormat,
+                DateTimeZoneHandling = DateTimeZoneHandling.Utc,
+                Error = delegate(object sender, Newtonsoft.Json.Serialization.ErrorEventArgs args)
+                {
+                    Errors.Add(args.ErrorContext.Error.Message);
+                    args.ErrorContext.Handled = true;
+                },
+                //Formatting = Formatting.Indented,
```
Comment (aaron-suarez) [inline Recurly/JsonSerializer.cs]: Same here, do we want to leave this in?

Comment (bhelx) [reply Recurly/JsonSerializer.cs]: Until we have better logging, this one is helpful to have on hand.

File: Recurly/BaseClient.cs (hunk 1 above)
Comment (bhelx) [reply Recurly/BaseClient.cs]: gonna remove this one

**Pull request merged**